src/fips.c

```diff
@@ -343,7 +392,7 @@ static void fips_lib_integrity_check(void)
 {
 	int rc;
 	char path[PATH_MAX];
-	const char *libname = "libica.so." VERSION;
+	const char *libname = LIBNAME ".so." VERSION;
```

Comment:
For a library that is `dlopen`ed, e.g., in ibmca engine, I find this check highly dangerous.  Assume you `dlopen` a libica you built yourself under `/home/myuser/libica.so` with the full path.  Then, this check will lead to an integrity check on the system installed libica which is not at all what will be used by the application.  I would instead suggest to `dl_iterate_phdr` over all PHDRs to search for the loaded library.  Of course, the `internal_test` is a major problem since it wants the fips-check to pass but does not even use the library and instead compiles all the files again.  But in this case, I guess the local compilation could also just add a preprocessor flag to bypass the fips check successfully.  Makes the code uglier, but should be the better solution.

Reply:
Just pushed your [patch 2/2] implementing the `dl_iterate_phdr` as discussed, thanks! We must not run the library integrity check for internal tests. 
I also adjusted the icainfo-cex output again to reflect that disabling some functionality via NO_CPACF or NO_SW_FALLBACKS outweights blocking an algorithm via fips build. So when in FIPS mode, icainfo-cex displays '-' rather than 'blocked'.